File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.java
class Solution {
    public int maxSubarrayLength(int[] nums, int k) {
        
        int ans = 0;
        HashMap<Integer,Integer> HM = new HashMap<>();
        for(int i=0,j=0;j<nums.length;)
        {
            if(HM.getOrDefault(nums[j],0) < k)
            {
               HM.put(nums[j],HM.getOrDefault(nums[j],0)+1);
               ans = Math.max(ans,j-i+1);
               j++;
            }
            else
            {
                HM.put(nums[i],HM.get(nums[i])-1);
                i++;
            }
        }
        return ans;
    }
}

File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        
       
        unordered_map<int,int>M;
        int ans = 0 , start = 0;
        for(int i=0;i<nums.size();i++)
        {
            M[nums[i]]++;

            while(M[nums[i]] > k)
                M[nums[start++]]--;
            
            ans = max(ans,i-start+1);
        }
        return ans;
    }
};